test/xspec-variable.xspec

```diff
@@ -5,29 +5,214 @@
+  <!-- Variables used only in this test file are in a dedicated namespace to prevent conflict with
+    global parameters or global variables in the file to be tested. -->
+  <t:variable name="myv:test-data" href="xspec-variable.xml"/>
+  <t:variable name="myv:test-3" select="$myv:test-data/*/test[xs:integer(@num) eq 3]"/>
+
+  <t:scenario label="Define variables in different ways">
+    <t:scenario label="@select only">
+      <t:variable name="myv:select-only" select="'y'"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{$myv:select-only}"/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode"/>
+      </t:call>
+      <t:expect label="the result" test="$x:result/self::bar/@role/string() eq 'y'"/>
+    </t:scenario>
+    <!--See xspec-variable_stylesheet.xspec for "@select and element content" scenario-->
+    <t:scenario label="@select and @href">
+      <t:variable name="myv:select-and-href" href="xspec-variable.xml" select="local-name(/test-data/*[1])"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{$myv:select-and-href}"/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode"/>
+      </t:call>
+      <t:expect label="the result" test="$x:result/self::bar/@role/string() eq 'test'"/>
+    </t:scenario>
+    <t:scenario label="element content">
+      <t:variable name="myv:content">y</t:variable>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{$myv:content}"/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode"/>
+      </t:call>
+      <t:expect label="the result" test="$x:result/self::bar/@role/string() eq 'y'"/>
+    </t:scenario>
+    <t:scenario label="element content, deeper">
+      <t:variable name="myv:content">
+        <xyz>y</xyz>
+      </t:variable>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{string($myv:content)}"/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode"/>
+      </t:call>
+      <t:expect label="the result" test="$x:result/self::bar/@role/string() eq 'y'"/>
+    </t:scenario>
+    <t:scenario label="@href only">
+      <t:variable name="myv:href-only" href="xspec-variable.xml"/>
+      <t:call function="my:foo">
+        <t:param name="elem"><foo role="{local-name($myv:href-only/test-data/*[1])}"/></t:param>
+        <t:param name="p" select="0"/>
+        <t:param name="textnode"/>
+      </t:call>
+      <t:expect label="the result" test="$x:result/self::bar/@role/string() eq 'test'"/>
+    </t:scenario>
+    <!-- REVIEW QUESTION: Remove the next scenario? A pending pull request on AirQuick's branch makes this pattern illegal. -->
```

Comment:
Re lines 77-88: Maybe after pull request 307 is merged in, I should remove the scenario from here, add content about variables to test/xspec-node-selection.xspec aligning with the approach used in that file, and update the bats files as needed. References: 3c01377b68fd9fea10ccaed6b6106e9372824098, d2fc95907f0f5c804b3ce8ef4800bac5fc6f7169.

Reply:
Yes, integrating it into `xspec-node-selection.xspec` will be very much appreciated.

Reply:
And #307 got merged just now.

Reply:
I made the updates I intended, but I made a mistake somewhere in the Git operations. I was able to update my master branch but had trouble getting those changes onto my "variablesupport" branch. I'll fix my commit in the next few days.

Reply:
I think I've managed to make Git happy with b5420c2!